pkg/factory/risingwave_object_factory.go

```diff
@@ -1201,19 +1201,25 @@ func (f *RisingWaveObjectFactory) NewMetaCloneSet(group string, podTemplates map
 
 	// Build the CloneSet
 	labelsOrSelectors := f.podLabelsOrSelectorsForGroup(consts.ComponentMeta, group)
-	metaCloneSet := &kruiseappsv1alpha1.CloneSet{
+	metaSts := &kruiseappsv1beta1.StatefulSet{
 		ObjectMeta: f.componentGroupObjectMeta(consts.ComponentMeta, group, true),
-		Spec: kruiseappsv1alpha1.CloneSetSpec{
+		Spec: kruiseappsv1beta1.StatefulSetSpec{
 			Replicas:       pointer.Int32(componentGroup.Replicas),
-			UpdateStrategy: buildUpgradeStrategyForCloneSet(componentGroup.UpgradeStrategy),
+			ServiceName:    f.componentName(consts.ComponentMeta, ""),
+			UpdateStrategy: buildUpgradeStrategyForAdvancedStatefulSet(componentGroup.UpgradeStrategy),
 			Selector: &metav1.LabelSelector{
 				MatchLabels: labelsOrSelectors,
 			},
-			Template: podTemplate,
+			Template:            podTemplate,
+			PodManagementPolicy: appsv1.ParallelPodManagement,
+			PersistentVolumeClaimRetentionPolicy: &kruiseappsv1beta1.StatefulSetPersistentVolumeClaimRetentionPolicy{
+				WhenDeleted: kruiseappsv1beta1.DeletePersistentVolumeClaimRetentionPolicyType,
+				WhenScaled:  kruiseappsv1beta1.DeletePersistentVolumeClaimRetentionPolicyType,
```

Comment:
What is the consequence of this? Will deleting the PVC Retention Policy Type also trigger a deletion of the PV? If so, do we want that? Does this mean that all pods loose their data if we scale the stateful set?

Reply:
Nope, it means 
- when deleting the `StatefulSet`, all PVCs from the PVC template will be deleted
- when scaling down the Pod of the `StatefulSet`, the corresponding PVC will also be deleted (and then the cloud controller should delete the PV as well)

You can check the details [here](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-retention).

For meta nodes, AFAIK, we won't have PVCs. Even if we have, that's for cache purposes and could be safely deleted.